File: lecture6/informationFlow.js
// metrics/informationFlow.js
function calculateInformationFlow(code) {
    const functionCallRegex = /\b\w+\s*\(.*\)/g;
    let fanIn = 0;
    let fanOut = 0;

    const matches = code.match(functionCallRegex);
    if (matches) {
        fanOut = matches.length;
        fanIn = matches.filter(match => match.includes('return')).length;  // Assumes fan-in is based on return statements
    }

    return { fanIn, fanOut, totalFlow: fanIn + fanOut };
}

module.exports = calculateInformationFlow;

File: lecture6/informationFlow.py
# metrics/information_flow.py

import re

def calculate_information_flow(code):
    function_call_pattern = r'\b\w+\s*\(.*?\)'

    matches = re.findall(function_call_pattern, code)

    fan_out = len(matches)
    fan_in = len([m for m in matches if 'return' in m])  # crude assumption

    return {
        'fan_in': fan_in,
        'fan_out': fan_out,
        'total_flow': fan_in + fan_out
    }
